src/V3InlineCFuncs.cpp

```diff
@@ -0,0 +1,266 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Inline small CFuncs into their callers
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2025 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+// V3InlineCFuncs's Transformations:
+//
+// For each CCall to a small CFunc:
+//   - Check if function is eligible for inlining (small enough, same scope)
+//   - Clone local variables with unique names to avoid collisions
+//   - Replace CCall with cloned function body statements
+//
+// Two tunables control inlining:
+//   --inline-cfuncs <n>         : Always inline if size <= n (default 20)
+//   --inline-cfuncs-product <n> : Also inline if size * call_count <= n (default 200)
+//
+//*************************************************************************
+
+#include "V3PchAstNoMT.h"  // VL_MT_DISABLED_CODE_UNIT
+
+#include "V3InlineCFuncs.h"
+
+#include "V3AstUserAllocator.h"
+#include "V3Stats.h"
+
+#include <map>
+#include <vector>
+
+VL_DEFINE_DEBUG_FUNCTIONS;
+
+//######################################################################
+// Helper visitor to check if a CFunc contains C statements
+// Uses clearOptimizable pattern for debugging
+
+class CFuncInlineCheckVisitor final : public VNVisitor {
```

Comment:
```suggestion
class CFuncInlineCheckVisitor final : public VNVisitorConst {
```

As not changing tree, performs better.  Call iterateChildrenConst.